web/oidc/oidc.go

```diff
@@ -375,35 +364,76 @@ func getToken(conf *Config, code string) (string, error) {
 }
 
 func getDomainFromUserInfo(conf *Config, token string) (string, error) {
-	req, err := http.NewRequest("GET", conf.UserInfoURL, nil)
+	if conf.AllowCustomInstance {
+		return "", errors.New("invalid configuration")
+	}
+	params, err := getUserInfo(conf, token)
 	if err != nil {
 		return "", err
 	}
+	return extractDomain(conf, params)
+}
+
+func checkDomainFromUserInfo(conf *Config, inst *instance.Instance, token string) error {
+	params, err := getUserInfo(conf, token)
+	if err != nil {
+		return err
+	}
+
+	if conf.AllowCustomInstance {
```

Comment:
Same here, quite strange to have to configure specifically the behavior with a setting.
Could be better to check if OAuth ID is available on settings and if not, to fallback to ID calculation.
This way, we allow easier implementation for future feature to enroll an existing Cozy to one (and even multiple) OAuth provider just by settings the OAuth ID.

Reply:
No, I really don't want to mix behaviors on the same context, as there are probably some bugs that will happen if we don't review carefully the code and modify some parts of it to manage this case.

Reply:
OK. Will be easy to change in the future if needed.